crates/wysiwyg/src/dom/dom_struct.rs

```diff
@@ -90,6 +90,27 @@ where
         self.document_mut().append_child(child)
     }
 
+    pub fn append_child_to_parent(
+        &mut self,
+        parent_handle: &DomHandle,
+        child: DomNode<S>,
+    ) -> DomHandle {
+        let parent = if let DomNode::Container(container) =
+            self.lookup_node_mut(parent_handle)
+        {
+            container
+        } else {
+            panic!("Parent is not a container node");
+        };
+        parent.append_child(child)
+    }
+
+    pub fn insert(&mut self, node_handle: &DomHandle, node: DomNode<S>) {
```

Comment:
Maybe insert_before would be a good name? It took me a while to figure out what this did. Maybe also a doc comment.